lib/src/widgets/alert_dialogs/alert_dialogs.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+
+import 'alert_dialog_impl.dart';
+
+/// Delete workout dialog
+@immutable
+class _DeleteDialog extends AlertDialogModel<bool> {
+  const _DeleteDialog({required String objName})
+      : super(
+          title: 'Delete $objName',
+          subtitle: 'Are you sure you want to delete this $objName?',
+          buttons: const {
+            'CANCEL': false,
+            'DELETE': true,
+          },
+        );
+}
+
+Future<bool> showDeleteWorkoutDialog(BuildContext context) {
+  return const _DeleteDialog(objName: 'workout').present(context).then(
+        (value) => value ?? false,
+      );
+}
+
+/// Add new exercise dialog
+@immutable
+class _AddNewExerciseDialog extends AlertDialogModel<bool> {
+  _AddNewExerciseDialog({required String objName, Widget? content})
+      : super(
+          title: 'New $objName',
+          subtitle:
+              'You can add a new exercise name or select from a existing one',
+          content: content,
+          buttons: const {
+            'CANCEL': false,
+            'ADD': true,
+          },
+        );
+}
+
+Future<bool> showAddNewExerciseDialog(BuildContext context, Widget? content) {
+  return _AddNewExerciseDialog(
+    objName: 'Exercise',
```

Comment:
I would maybe have the objName as an optional parameter to the showAddNewExerciesDialog method because otherwise I don't really see the reason it isn't just hardcoded in the constructor above, same for the rest of the dialogs.